### PR TITLE
Improve logging in project deletes

### PR DESCRIPTION
**Pull request by saipavan10-git**
This pull request includes significant changes to the `delete_project` function and its associated tests. The changes enhance logging, and update the test cases to reflect the new logging structure. Addresses #169 

I tested the new delete function and now the project delete is visible on REDCap UI.

image

## Enhancements to `delete_project` function:

* Enhanced logging by adding detailed log parameters such as `ts`, `user`, `ip`, `page`, `event`, `object_type`, `sql_log`, `pk`, `event_id`, `data_values`, `description`, `legacy`, and `change_reason`.

### CODE REVIEW

File: R/delete_project.R [hunk 1]
```diff
@@ -34,47 +33,86 @@ delete_project <- function(project_id, conn) {
...
+      },
+      error = function(error_message) {
+        print(error_message)
+        return(FALSE)
+      }
+    )
+
+    # Define logging parameters
+    ts <- format(Sys.time(), "%Y%m%d%H%M%S") # Time stamp
+    user <- "admin" # Placeholder for user
```
Comment (pbchase) [inline R/delete_project.R:63]: Test this with an invalid username to see how REDCap's UI will behave.

Then, revise the code to default it first to, the value in sys.getenv("API_USER"), then--if invalid usernames are allowed--the script name.

File: R/delete_project.R [hunk 2]
```diff
@@ -34,47 +33,86 @@ delete_project <- function(project_id, conn) {
...
+    ip <- "192.168.65.1" # Placeholder for IP address
```
Comment (pbchase) [inline R/delete_project.R:64]: Use getip::getip("local") to get the local IP address. Make getip a required package for redcapcustodian.

**Review by pbchase — CHANGES_REQUESTED** [review]
See comments

**Comment by pbchase** [comment]
@saipavan10-git, note that the checks are failing. If you drill in on that error you will see the error is pretty obvious: redcapcustodian now needs `getip` and the container used by the github action does not have that. I've fixed these before, but I should hand this one to you. I'll show you path, so you can fix it. 

Could you document this procedure in addition to fixing it? The procedure doesn't seem clear unless you've already set up GitHub actions to perform automated testing. Any thoughts on where to put this procedure? Could we add a vignette for developers?

You can see the configuration for the action that failed at [run-tests.yaml](https://github.com/ctsit/redcapcustodian/blob/master/.github/workflows/run-tests.yaml). It uses an image at [ghcr.io/ctsit/rstudio-ci](ghcr.io/ctsit/rstudio-ci). The Dockerfile that defines that is at [Dockerfile](https://github.com/ctsit/rstudio-dockerfile/blob/master/Dockerfile). Historically, we have added required packages in this Dockerfile. That should work quickly and easily, but I think you can install packages in `run-tests.yaml` that might be a better  way to make the package self-documenting, but that might also make tests slower to run.

**Comment by saipavan10-git** [comment]
I updated this [dockerfile](https://github.com/ctsit/redcapcustodian/blob/da6185fdb37bcece14e1dba7df1863665bd78e80/Dockerfile) thinking this will fix the issue.

Adding a guide vignette for developers is a great idea. I can create one to document the fix for this issue. If we add new dependencies more frequently, we could add a step to setup dependencies in run-tests.yaml like in [pkgdown.yaml](https://github.com/ctsit/redcapcustodian/blob/da6185fdb37bcece14e1dba7df1863665bd78e80/.github/workflows/pkgdown.yaml#L31-L34). This will reduce the need to update the [rstudio-Dockerfile](https://github.com/ctsit/rstudio-dockerfile/blob/master/Dockerfile). Also, setup r dependencies has an option to [cache](https://arc.net/l/quote/xuqtunbp) the packages that might help with the speed of the gh action.

**Comment by pbchase** [comment]
> I updated this [dockerfile](https://github.com/ctsit/redcapcustodian/blob/da6185fdb37bcece14e1dba7df1863665bd78e80/Dockerfile) thinking this will fix the issue.

That's necessary, but not sufficient to make CI-tests work

> Adding a guide vignette for developers is a great idea. I can create one to document the fix for this issue. If we add new dependencies more frequently, we could add a step to setup dependencies in run-tests.yaml like in [pkgdown.yaml](https://github.com/ctsit/redcapcustodian/blob/da6185fdb37bcece14e1dba7df1863665bd78e80/.github/workflows/pkgdown.yaml#L31-L34). This will reduce the need to update the [rstudio-Dockerfile](https://github.com/ctsit/rstudio-dockerfile/blob/master/Dockerfile). 

Right! I knew we had done this before, but I couldn't locate it.

> Also, setup r dependencies has an option to [cache](https://arc.net/l/quote/xuqtunbp) the packages that might help with the speed of the gh action.

That minimizes the downside of the more local management of dependencies.  I like it :-)

**Review by pbchase — APPROVED** [review]

**Pull request merged**